src/sentry/data_export/tasks.py

```diff
@@ -21,157 +34,206 @@
+    processor = DiscoverProcessor(
+        discover_query=data_export.query_info, organization_id=data_export.organization_id,
+    )
+    raw_data_unicode = processor.data_fn(limit=limit, offset=offset)["data"]
+    # TODO(python3): Remove next line once the 'csv' module has been updated to Python 3
+    # See associated comment in './utils.py'
+    raw_data = convert_to_utf8(raw_data_unicode)
+    raw_data = processor.handle_fields(raw_data)
+    return processor, raw_data
+
+
+@transaction.atomic()
+def store_export_chunk_as_blob(data_export, bytes_written, fileobj, blob_size=DEFAULT_BLOB_SIZE):
+    # adapted from `putfile` in  `src/sentry/models/file.py`
+    bytes_offset = 0
+    while True:
+        contents = fileobj.read(blob_size)
+        if not contents:
+            return bytes_offset
+
+        blob_fileobj = ContentFile(contents)
+        blob = FileBlob.from_file(blob_fileobj, logger=logger)
+        ExportedDataBlob.objects.create(
+            data_export=data_export, blob=blob, offset=bytes_written + bytes_offset
         )
-    except ExportError as error:
-        metrics.incr("dataexport.error", tags={"error": six.text_type(error)}, sample_rate=1.0)
-        logger.info("dataexport.error: {}".format(six.text_type(error)))
-        capture_exception(error)
-        raise error
 
-    writer = create_writer(file, processor.header_fields)
-    iteration = 0
-    is_completed = False
-    while not is_completed:
-        offset = batch_size * iteration
-        next_offset = batch_size * (iteration + 1)
-        is_exceeding_limit = export_limit and export_limit < next_offset
-        raw_data_unicode = processor.data_fn(offset=offset, limit=batch_size)["data"]
-        # TODO(python3): Remove next line once the 'csv' module has been updated to Python 3
-        # See associated comment in './utils.py'
-        raw_data = convert_to_utf8(raw_data_unicode)
-        raw_data = processor.handle_fields(raw_data)
-        if is_exceeding_limit:
-            # Since the next offset will pass the export_limit, just write the remainder
-            writer.writerows(raw_data[: export_limit % batch_size])
-        else:
-            writer.writerows(raw_data)
-            iteration += 1
-        # If there are no returned results, or we've passed the export_limit, stop iterating
-        is_completed = len(raw_data) == 0 or is_exceeding_limit
+        bytes_offset += blob.size
 
+        # there is a maximum file size allowed, so we need to make sure we don't exceed it
+        if bytes_written + bytes_offset >= MAX_FILE_SIZE:
+            transaction.set_rollback(True)
+            return 0
+
+
+@instrumented_task(name="sentry.data_export.tasks.merge_blobs", queue="data_export")
+def merge_export_blobs(data_export_id, **kwargs):
+    try:
+        data_export = ExportedData.objects.get(id=data_export_id)
+    except ExportedData.DoesNotExist as error:
+        capture_exception(error)
```

Comment:
Should we just `raise` here?

Reply:
what difference will `raise` have here? does an exception from a task get handled different somewhere?

Reply:
No it'd be the same thing, just a bit more concise. Tbh if all we're doing here is capturing the error and returning we could just remove the try/catch and let the error bubble up, it'll have the same effect in sentry. Nbd either way though.